### PR TITLE
feat(bigquery): make rowIterator._to_dataframe_iterable public

### PR DESCRIPTION
**Pull request by HemangChothani**
IPR 7339

### CODE REVIEW

File: bigquery/google/cloud/bigquery/table.py [hunk 1]
```diff
@@ -1554,11 +1554,44 @@ def to_arrow(
...
+        Returns:
+            pandas.DataFrame:
+                A generator of :class:`~pandas.DataFrame`.
+
+        Raises:
+            ValueError:
+                If the :mod:`pandas` library cannot be imported.
         """
+        if pandas is None:
+            raise ValueError(_NO_PANDAS_ERROR)
```
Comment (IlyaFaer) [inline bigquery/google/cloud/bigquery/table.py]: I'd say this is a `RuntimeError`, but I see that in BigQuery they're using `ValueError` instead. Okay

File: bigquery/google/cloud/bigquery/table.py [hunk 2]
```diff
@@ -1554,11 +1554,44 @@ def to_arrow(
...
+        if dtypes is None:
+            dtypes = {}
```
Comment (IlyaFaer) [inline bigquery/google/cloud/bigquery/table.py]: Not sure about this line. Why not to set `dtypes` to `{}` by default in func definition?

Comment (HemangChothani) [reply bigquery/google/cloud/bigquery/table.py]: @IlyaFaer I also wonder why they didn't set  `{}`  as a default value in other functions, like `to_daframe()` .function.

File: bigquery/google/cloud/bigquery/table.py (hunk 1 above)
Comment (HemangChothani) [reply bigquery/google/cloud/bigquery/table.py]: @IlyaFaer  Yes they are using `ValueError` for the import error.

File: bigquery/google/cloud/bigquery/table.py (hunk 2 above)
Comment (IlyaFaer) [reply bigquery/google/cloud/bigquery/table.py]: Oh, you've taken it from another func? Okay then
Still, I would ask them in PR, why are they doing that way, Maybe they just didn't paid attention

**Pull request closed without merging**